module-4/backend/week_one.md

```diff
@@ -1,21 +1,63 @@
 ## Week One - Module 4 Recap
 
-Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week. 
+Fork this respository. Answer the questions to the best of your ability. Try to answer them with limited amount of external research. These questions cover the majority of what we've learned this week.
 
-Note: When you're done, submit a PR. 
+Note: When you're done, submit a PR.
 
 1. What's the most useful thing you learned from completing the intermission week work?
+
+  How to make a for loop; however, I wish there were more basic exercises to help us learn by repetition.
+
 2. What are some tools to help debug JavaScript code?
+
+  Debugger, pryjs and using console.log()
+
 3. What are some tools you need in order to unit test your JavaScript?
+
+  Mocha and chai.
+
 4. What is the syntax for invoking a function?
+
+  ```javascript
+  function();
+  ```
+
 5. What's the difference between `==` and `===` in JavaScript?
-6. What's the difference between asynchronous and synchronous JavaScript? 
+
+  `==` is used for checking equality and `===` is checking for a true/false result
+
+6. What's the difference between asynchronous and synchronous JavaScript?
+
+  In asynchronous JS the code is executed without an specific order and while long, slow code is being executed, a smaller and faster one will execute a the same time; in synchronous JS, the code is executed in order and the following command/code will only be executed after the prior code is finished.
+
 7. What's a callback function and what are some reasons when we use/need callback functions?
+
+  A callback function is a function that is passed to another function as a parameter. The callback function can be executed inside that other function.
+
 8. What's the biggest difference between a promise and a callback?
+
```

Comment:
i suggest doing some research with this! it's important to know both of these and what they can do for you. comparing/contrasting may not be as important